applications/blackhole/src/modules/bh_call.erl

```diff
@@ -22,63 +22,64 @@
                    ]).
 
 -spec handle_event(bh_context:context(), kz_json:object()) -> 'ok'.
-handle_event(Context, EventJObj) ->
+handle_event(#bh_context{binding=Binding} = Context, EventJObj) ->
     kz_util:put_callid(EventJObj),
     lager:debug("handle_event fired for ~s ~s", [bh_context:account_id(Context), bh_context:websocket_session_id(Context)]),
-    'true' = kapi_call:event_v(EventJObj)
-        andalso is_account_event(Context, EventJObj),
+    'true' = kapi_call:event_v(EventJObj),
     lager:debug("valid event and emitting to ~p: ~s", [bh_context:websocket_pid(Context), event_name(EventJObj)]),
-    blackhole_data_emitter:emit(bh_context:websocket_pid(Context), event_name(EventJObj), EventJObj).
-
-is_account_event(Context, EventJObj) ->
-    kz_json:get_first_defined([<<"Account-ID">>
-                              ,[<<"Custom-Channel-Vars">>, <<"Account-ID">>]
-                              ], EventJObj
-                             ) =:=
-        bh_context:account_id(Context).
+    NormJObj = kz_json:normalize_jobj(kz_json:set_value(<<"Binding">>, Binding, EventJObj)),
```

Comment:
> is this not related to KAZOO-4916
It is related. Binding value is unique in connection and therefore is an "unique key". There is no need to generate unique key on unique key. And what @jr actually wants is to have binding value in each event.